cli/cliui/prompt.go

```diff
@@ -79,7 +79,9 @@ func Prompt(inv *clibase.Invocation, opts PromptOptions) (string, error) {
 
 	errCh := make(chan error, 1)
 	lineCh := make(chan string)
+
 	go func() {
+		defer close(lineCh)
```

Comment:
We shouldn’t close here, we’re not checking channel closure below so this could lead to a bug. Or did closing the channel have some benefit I’m not considering?

Reply:
It's more of a future-proofing. In general, I like to close every channel so that bad receives crash instead of hang.

Reply:
It's more of a future-proofing. In general, I like to close every channel when it we're certain there will be no more sends so that bad receives crash instead of hang.

Reply:
In cases like these where we're relying on other closure signals, like the context, closing the channel does not seem safe. AFAICT it wouldn't crash either, it'd successfully pass along the empty string. (I.e. a send would crash, but we're only sending from the goroutine that closes it so there's no benefit.)

Reply:
Empty string is true, and I guess there's a question of whether the empty string is better than a hang? I removed it for now in deference.

Reply:
A hand would most certainly be better in this case because an empty string would indicate success in this instance, suggesting the user gave that input.